File: src/main/java/net/atos/client/bag/util/TypeOpenbareRuimteEnumAdapter.java
/*
 * SPDX-FileCopyrightText: 2023 Atos
 * SPDX-License-Identifier: EUPL-1.2+
 */

package net.atos.client.bag.util;

import jakarta.json.bind.adapter.JsonbAdapter;

import nl.info.client.bag.model.generated.TypeOpenbareRuimte;

public class TypeOpenbareRuimteEnumAdapter implements JsonbAdapter<TypeOpenbareRuimte, String> {

    @Override
    public String adaptToJson(final TypeOpenbareRuimte typeOpenbareRuimte) {
        return typeOpenbareRuimte.toString();
    }

    @Override
    public TypeOpenbareRuimte adaptFromJson(final String json) {
        return TypeOpenbareRuimte.fromValue(json);
    }
}

File: src/main/kotlin/nl/info/client/bag/util/TypeOpenbareRuimteEnumAdapter.kt
/*
 * SPDX-FileCopyrightText: 2023 Atos, 2026 INFO.nl
 * SPDX-License-Identifier: EUPL-1.2+
 */
package nl.info.client.bag.util

import jakarta.json.bind.adapter.JsonbAdapter
import nl.info.client.bag.model.generated.TypeOpenbareRuimte

class TypeOpenbareRuimteEnumAdapter : JsonbAdapter<TypeOpenbareRuimte, String> {
    override fun adaptToJson(typeOpenbareRuimte: TypeOpenbareRuimte): String = typeOpenbareRuimte.toString()
    override fun adaptFromJson(json: String): TypeOpenbareRuimte = TypeOpenbareRuimte.fromValue(json)
}
